exporters/mock/include/opentelemetry/exporters/mock/mock_span_exporter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace mock
+{
+/**
+ * A mock exporter that switches a flag once a valid recordable was received.
+ */
+class MockSpanExporter final : public opentelemetry::sdk::trace::SpanExporter
+{
+public:
+  MockSpanExporter(
+      std::shared_ptr<std::vector<std::unique_ptr<sdk::trace::SpanData>>> spans_received,
+      std::shared_ptr<bool> shutdown_called = std::shared_ptr<bool>(new bool(false)));
```

Comment:
Let's call this `InMemorySpanExporter` instead of `MockSpanExporter`.

Reply:
Also, let's wrap `spans_received` and `shutdown_called` in a separate class (e. g. `InMemorySpanData`) that provides thread-safe access to the data. The current solution makes it impossible to access `spans_received` in a thread-safe manner.

Ideally, we would use the lock-free [CircularBuffer](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/sdk/src/common/circular_buffer.h), as it provides both thread safe access and the ability to manage a maximum capacity. An alternative implementation would be a vector with locks around each read and write.

Reply:
Updated to InMemorySpanExporter and updated functionality to use CircularBuffer